templates/deployments.yaml

```diff
@@ -1242,6 +1326,10 @@ spec:
         env:
         - name: ST2CLIENT
           value: "1"
+{{- range $env, $value := .Values.st2client.env }}
+        - name: {{ $env | quote }}
+          value: {{ $value | quote }}
+{{- end }}
```

Comment:
A couple of jobs also have an `env:` block already. So, we could split the template into two to accommodate this usage.
```
{{- define "custom-env-entries" -}}
  {{- range $env, $value := .env }}
- name: {{ $env | quote }}
  value: {{ $value | quote }}
  {{- end }}
{{- end -}}
{{- define "custom-env" -}}
  {{- if .env }}
env:
{{ include "custom-env-entries" . }}
  {{- end }}
{{- end -}}
```
Which would make this block be:
```
{{ include "custom-env-entries" .Values.st2client }}
```